test/base/test_buffer.py

```diff
@@ -320,35 +410,204 @@ def test_hdf5():
         assert len(_buffers[k]) == len(buffers[k])
         assert np.allclose(_buffers[k].act, buffers[k].act)
         assert _buffers[k].stack_num == buffers[k].stack_num
-        assert _buffers[k]._maxsize == buffers[k]._maxsize
-        assert _buffers[k]._index == buffers[k]._index
+        assert _buffers[k].maxsize == buffers[k].maxsize
         assert np.all(_buffers[k]._indices == buffers[k]._indices)
     for k in ["array", "prioritized"]:
+        assert _buffers[k]._index == buffers[k]._index
         assert isinstance(buffers[k].get(0, "info"), Batch)
         assert isinstance(_buffers[k].get(0, "info"), Batch)
     for k in ["array"]:
         assert np.all(
             buffers[k][:].info.number.n == _buffers[k][:].info.number.n)
         assert np.all(
             buffers[k][:].info.extra == _buffers[k][:].info.extra)
+    # check shallow copy in ReplayBuffers
+    for k in ["vector", "cached"]:
+        buffers[k].info.number.n[0] = -100
+        assert buffers[k].buffers[0].info.number.n[0] == -100
+    # check if still behave normally
+    for k in ["vector", "cached"]:
+        kwargs = {
+            'obs': Batch(index=np.array([5])),
+            'act': 5,
+            'rew': np.array([2, 1]),
+            'done': False,
+            'info': {"number": {"n": i}, 'Timelimit.truncate': True},
+        }
+        buffers[k].add(**Batch.cat([[kwargs], [kwargs], [kwargs], [kwargs]]))
+        act = np.zeros(buffers[k].maxsize)
+        if k == "vector":
+            act[np.arange(5)] = np.array([0, 1, 2, 3, 5])
+            act[np.arange(5) + size] = np.array([0, 1, 2, 3, 5])
+            act[np.arange(5) + size * 2] = np.array([0, 1, 2, 3, 5])
+            act[size * 3] = 5
+        elif k == "cached":
+            act[np.arange(9)] = np.array([0, 1, 2, 0, 1, 2, 0, 1, 2])
+            act[np.arange(3) + size] = np.array([3, 5, 2])
+            act[np.arange(3) + size * 2] = np.array([3, 5, 2])
+            act[np.arange(3) + size * 3] = np.array([3, 5, 2])
+            act[size * 4] = 5
+        assert np.allclose(buffers[k].act, act)
 
     for path in paths.values():
         os.remove(path)
 
     # raise exception when value cannot be pickled
-    data = {"not_supported": lambda x: x*x}
+    data = {"not_supported": lambda x: x * x}
     grp = h5py.Group
     with pytest.raises(NotImplementedError):
         to_hdf5(data, grp)
     # ndarray with data type not supported by HDF5 that cannot be pickled
-    data = {"not_supported": np.array(lambda x: x*x)}
+    data = {"not_supported": np.array(lambda x: x * x)}
     grp = h5py.Group
     with pytest.raises(RuntimeError):
         to_hdf5(data, grp)
 
 
+def test_vectorbuffer():
+    buf = ReplayBuffers([ReplayBuffer(size=5) for i in range(4)])
```

Comment:
I would rather split this test, for the same reason that the previous one. It is no longer clear what you are checking, making hard to maintain and update if the specification changes.

Reply:
I'll re-organize this unit test. I think it is messy too. That's my bad.